django/utils/autoreload.py

```diff
@@ -387,7 +387,7 @@ def _watch_root(self, root):
                 logger.warning('Unable to watch root dir %s as neither it or its parent exist.', root)
                 return
             root = root.parent
-        result = self.client.query('watch-project', str(root.absolute()))
+        result = pywatchman.client(timeout=5.0).query('watch-project', str(root.absolute()))
```

Comment:
@Jacobkg the place to make this change would be in the definition of the `client` property:

```python
    @cached_property
    def client(self):
        return pywatchman.client(timeout=5.0)
```

but neither place affects the client used to test for `pywatchman` availability, inside the `check_availability` method:

```
        client = pywatchman.client(timeout=0.01)
```

Reply:
@beaugunderson Thanks for the feedback! Keep in mind I didn't write any of the original code and can only guess the original intention based on what is already there. Given that:

In `check_availability` we are seeing whether Watchman is installed on the machine at all, and thus want to have an extremely low timeout (`0.01`) so we aren't stuck waiting in the case it is not there. Presumably that small timeout is sufficient for the purposes of checking availability and I wouldn't want to change it. I certainly haven't found it to be an issue nor is that the subject of the ticket in question

I will make the other two suggested changes that you proposed

Reply:
the call with the `0.01` timeout in `check_availability` does fail for me sometimes on startup :)

Reply:
@beaugunderson Ah! Do you have a sense for what timeout works for you with high probability?

@orf I was wondering if as the original author of the Watchman autoloader you had thoughts on:
1. This PR - raising the timeout to `5.0 seconds` in general
2. Whether it is a bad idea to increase the `check_availability` timeout above `0.1` seconds?

Reply:
I think a timeout of `0.1` for `check_availability` is reasonable, I chose `0.01` as I assumed that was a reasonable default for a `VERSION` request to a local service.

If it's not, then I'm happy with a `0.1` or `0.2` value.

Regarding changing the timeout to 5 seconds, it seems like this is a bit of guesswork. Would it be better to control this through an environment variable? I'm happy to up the default timeout to 5 seconds, but it might become a game of cat and mouse if anyone with larger projects complain.

Reply:
@orf I agree completely that we don't want to be continually tweaking the timeout. My instinct is to try out 5 and if people keep having problems then we add the environment variable. However, you have a lot more experience contributing to Django so I defer to your opinion if you feel like going a different way.

Reply:
I think seeing as we are changing this once it makes sense to make it future proof. The work in this case is exactly the same, it's a matter of adding `os.environ.get('DJANGO_WATCHMAN_TIMEOUT', 5)` instead of just `5`, which seems pretty safe to do. We'd have to add a small test and some documentation as well.

If you're happy to work on this I can review it and give feedback, but if you've got time constraints I can pick it up 👍

Reply:
@orf Thank you for the encouragement. I'm feeling in a bit over my head as far as the tests and documentation but I'm going to give it a try! I'll ping you when its ready for review